server.go

```diff
@@ -286,6 +294,28 @@ func (s *Server) handleProxyError(err error, w http.ResponseWriter) {
 	w.Write([]byte("Internal Server Error"))
 }
 
+func (s *Server) GetRoute(ctx context.Context) *Route {
+	if ctx == nil {
+		return nil
+	}
+
+	if route := ctx.Value(routeContextKey{}); route != nil {
+		return route.(*Route)
+	}
+	return nil
+}
+
+func (s *Server) GetParameters(ctx context.Context) map[string]string {
```

Comment:
```suggestion
func ParametersFromContext(ctx context.Context) map[string]string {
```

Similar to above, thoughts on this naming?